contracts/src/CrossChainApplications/examples/NativeTokenBridge/NativeTokenDestination.sol

```diff
@@ -213,27 +280,47 @@ contract NativeTokenDestination is TeleporterOwnerUpgradeable, INativeTokenDesti
 
         (address recipient, uint256 amount) = abi.decode(message, (address, uint256));
         require(recipient != address(0), "NativeTokenDestination: zero recipient address");
-        require(amount != 0, "NativeTokenDestination: zero transfer value");
+
+        uint256 scaledAmount = _scaleTokens(amount, true);
+        require(scaledAmount >= 0, "NativeTokenDestination: zero scaled amount received");
```

Comment:
I don't think this require is necessary since a `uint256` is always >= 0. If the scaled amount is 0, we won't call mint but still emit an event with an amount of 0, which I think is the correct behavior.